File: leetcode/772/772.rb
# Given the head of a singly linked list, return true if it is a palindrome.

# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {Boolean}
def is_palindrome(head)
    list = to_list(head)
    verify_palindrome(list)
end

def to_list(node)
    list = []
    while node != nil 
        list << node
        node = node.next
    end
    list
end

def verify_palindrome(list)
    first = 0
    last = list.length - 1
    while first <= last
        return false if list[first].val != list[last].val
        first += 1
        last -= 1
    end
    true
end

File: leetcode/772/772.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        '''
        Given the head of a singly linked list, return true if it is a palindrome.
        '''
        
        # convert to list
        asList = []
        while head is not None:
            result.append(head.val)
            head = head.next
            
        # check if list is palindromic    
        for i in range(0, len(asList) // 2):
            if asList[i] != asList[-(i + 1)]:
                return False
        return True
    
        
